lib/agent.js

```diff
@@ -60,20 +60,41 @@ class Agent extends EventEmitter {
     return pool
   }
 
-  close () {
+  close (callback) {
+    if (typeof callback !== 'undefined' && typeof callback !== 'function') {
+      throw new InvalidArgumentError('must be a function')
+    }
     const closePromises = []
     for (const pool of this[kCache].values()) {
       closePromises.push(pool.close())
     }
-    return Promise.all(closePromises)
+    if (callback === undefined) {
+      return Promise.all(closePromises)
+    }
+
+    Promise.all(closePromises)
+      .then(() => process.nextTick(callback))
+      .catch((err) => process.nextTick(callback, err))
```

Comment:
Please fix conflicts. Also the tests seem a bit weird. Can you just add tests calling close/abort and making sure the callback gets invoked? Doesn't have to be anything fancy.